src/pybind/mgr/dashboard/frontend/src/app/shared/forms/cd-validators.ts

```diff
@@ -683,4 +684,30 @@ export class CdValidators {
       return { invalidAddress: !(addressTest && portTest) };
     };
   }
+
+  /**
+   * Validator function to validate endpoints, allowing FQDN, IPv4, and IPv6 addresses with ports.
+   * Accepts multiple endpoints separated by commas.
+   */
+  static endpointValidator(control: AbstractControl): ValidationErrors | null {
```

Comment:
this is not an endpoint validator btw as endpoint refers to the path of the API. I believe this is validating a url (or a base url that simply has `<protocol>://<ip>:<port>`. Let's call it that.
```suggestion
  static url(control: AbstractControl): ValidationErrors | null {
```